From fb507124f8bb4f93a1e9ebc28673d0faaee4a18c Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:51 -0400
Subject: [tools] perf/tests: Show the inner make output when an error happens

Message-id: <1410181752-25631-287-git-send-email-jolsa@redhat.com>
Patchwork-id: 91886
O-Subject: [PATCH RHEL7.1 BZ1134356 286/307] perf tests: Show the inner make output when an error happens
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit a5c5009f72ced4b90f8715137784c0cd8b244544
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Thu Jun 5 17:15:47 2014 -0300

description
===========
Before:

  [acme@zoo linux]$ make -C tools/perf -f tests/make make_static
  make: Entering directory `/home/git/linux/tools/perf'
  - make_static: cd . && make -f Makefile DESTDIR=/tmp/tmp.JcWuM4Zu9f LDFLAGS=-static
  make: *** [make_static] Error 1
  make: Leaving directory `/home/git/linux/tools/perf'
  [acme@zoo linux]$

After:

  [acme@zoo linux]$ make -C tools/perf -f tests/make make_static
  make: Entering directory `/home/git/linux/tools/perf'
  - make_static: cd . && make -f Makefile DESTDIR=/tmp/tmp.X3su83i14u LDFLAGS=-static
  cd . && make -f Makefile DESTDIR=/tmp/tmp.X3su83i14u LDFLAGS=-static
    BUILD:   Doing 'make -j4' parallel build
  config/Makefile:303: *** No static glibc found, please install glibc-static.  Stop.
  make[1]: *** [all] Error 2
    test: test -x ./perf
  make: Leaving directory `/home/git/linux/tools/perf'
  [acme@zoo linux]$
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/tests/make b/tools/perf/tests/make
index 2f92d6e..69a71ff 100644
--- a/tools/perf/tests/make
+++ b/tools/perf/tests/make
@@ -205,8 +205,7 @@ $(run):
  ( eval $$cmd ) >> $@ 2>&1; \
  echo "  test: $(call test,$@)" >> $@ 2>&1; \
  $(call test,$@) && \
- rm -f $@ \
- rm -rf $$TMP_DEST
+ rm -rf $@ $$TMP_DEST || (cat $@ ; false)
 
 $(run_O):
  $(call clean)
@@ -217,9 +216,7 @@ $(run_O):
  ( eval $$cmd ) >> $@ 2>&1 && \
  echo "  test: $(call test_O,$@)" >> $@ 2>&1; \
  $(call test_O,$@) && \
- rm -f $@ && \
- rm -rf $$TMP_O \
- rm -rf $$TMP_DEST
+ rm -rf $@ $$TMP_O $$TMP_DEST || (cat $@ ; false)
 
 tarpkg:
  @cmd="$(PERF)/tests/perf-targz-src-pkg $(PERF)"; \
-- 
1.7.1